SOURCES/0001-losetup-Add-support-for-logical-block-size.patch

```diff
@@ -0,0 +1,279 @@
+ 	return rc;
+ }
+ 
++/*
++ * @lc: context
++ * @blocksize: returns logical blocksize for the given device
++ *
++ * Returns: <0 on error, 0 on success
++ */
++int loopcxt_get_blocksize(struct loopdev_cxt *lc, uint64_t *blocksize)
++{
++	struct sysfs_cxt *sysfs = loopcxt_get_sysfs(lc);
++	int rc = -EINVAL;
++
++	if (sysfs)
++		rc = sysfs_read_u64(sysfs, "queue/logical_block_size", blocksize);
++
++	/* Fallback based on BLKSSZGET ioctl */
++	if (rc) {
++		int fd = loopcxt_get_fd(lc);
++		int sz = 0;
++
++		if (fd < 0)
++			return -EINVAL;
++		rc = blkdev_get_sector_size(fd, &sz);
++		if (rc)
++			return rc;
++
++		*blocksize = sz;
++	}
++
++	DBG(CXT, ul_debugobj(lc, "get_blocksize [rc=%d]", rc));
++	return rc;
++}
++
+ /*
+  * @lc: context
+  * @sizelimit: returns size limit for the given device
+@@ -1288,6 +1320,24 @@ int loopcxt_set_capacity(struct loopdev_cxt *lc)
+ 	return 0;
+ }
+ 
++int loopcxt_set_blocksize(struct loopdev_cxt *lc, unsigned long blocksize)
++{
++	int fd = loopcxt_get_fd(lc);
++
++	if (fd < 0)
++		return -EINVAL;
++
++	/* Kernels prior to v4.14 don't support this ioctl */
++	if (ioctl(fd, LOOP_SET_BLOCK_SIZE, blocksize) < 0) {
++		int rc = -errno;
++		DBG(CXT, ul_debugobj(lc, "LOOP_SET_BLOCK_SIZE failed: %m"));
++		return rc;
++	}
++
++	DBG(CXT, ul_debugobj(lc, "logical block size set"));
++	return 0;
++}
++
+ int loopcxt_delete_device(struct loopdev_cxt *lc)
+ {
+ 	int fd = loopcxt_get_fd(lc);
+diff --git a/sys-utils/losetup.8 b/sys-utils/losetup.8
+index 9a8c1d597..ee704ca81 100644
+--- a/sys-utils/losetup.8
++++ b/sys-utils/losetup.8
+@@ -102,6 +102,8 @@ device
+ specify which columns are to be printed for the \fB\-\-list\fP output
+ .IP "\fB\-\-sizelimit \fIsize\fP"
+ the data end is set to no more than \fIsize\fP bytes after the data start
++.IP "\fB-b, \-\-logical-blocksize \fIsize\fP"
++Set the logical block size of the loop device in bytes (since Linux 4.14).
+ .IP "\fB\-P, \-\-partscan\fP"
+ force kernel to scan partition table on newly created loop device
+ .IP "\fB\-r, \-\-read-only\fP"
+diff --git a/sys-utils/losetup.c b/sys-utils/losetup.c
+index d9b0c9b61..6e547fea8 100644
+--- a/sys-utils/losetup.c
++++ b/sys-utils/losetup.c
+@@ -33,6 +33,7 @@ enum {
+ 	A_SHOW_ONE,		/* print info about one device */
+ 	A_FIND_FREE,		/* find first unused */
+ 	A_SET_CAPACITY,		/* set device capacity */
++	A_SET_BLOCKSIZE,	/* set logical block size of the loop device */
+ };
+ 
+ enum {
+@@ -46,6 +47,7 @@ enum {
+ 	COL_PARTSCAN,
+ 	COL_RO,
+ 	COL_SIZELIMIT,
++	COL_BLOCKSIZE,
+ };
+ 
+ struct tt *tt;
+@@ -68,6 +70,7 @@ static struct colinfo infos[] = {
+ 	[COL_RO]          = { "RO",           1, TT_FL_RIGHT, N_("read-only device")},
+ 	[COL_SIZELIMIT]   = { "SIZELIMIT",    5, TT_FL_RIGHT, N_("size limit of the file in bytes")},
+ 	[COL_MAJMIN]      = { "MAJ:MIN",      3, 0, N_("loop device major:minor number")},
++	[COL_BLOCKSIZE]   = { "BLOCKSIZE",    4, SCOLS_FL_RIGHT, N_("logical block size in bytes")},
```

Comment:
`SCOLS_FL_RIGHT` is undefined. Might have be renamed and the right value seems to be `TT_FL_RIGHT`.

Reply:
see below